pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -225,8 +225,7 @@ func validateMetric(annotations map[string]string) *apis.FieldError {
 				return nil
 			}
 		case HPA:
-			switch metric {
-			case CPU, Memory:
+			if metric != "" {
```

Comment:
I don't think this is very important (and pretty subjective, so please feel free to disregard this comment), but stylistically, using a `switch` here looks a little nicer to me (as it mirrors the KPA case above), i.e.

```golang
case HPA:
	switch metric {
	case "":
		break
	default:
		return nil
	}
```

Reply:
I understand, I will make that change.